### PR TITLE
[SPARK-35056][SQL] Group exception messages in execution/streaming

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR group exception messages in `sql/core/src/main/scala/org/apache/spark/sql/execution/streaming`.

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 1]
```diff
@@ -1329,4 +1330,96 @@ object QueryExecutionErrors {
...
+    new IOException(s"Failed to rename temp file $srcPath to $dstPath as rename returned false")
+  }
+
+  def legacyMetadataPathExistsError(metadataPath: Path, legacyMetadataPath: Path): Throwable = {
+    new SparkException(
+      s"""
+         |Error: we detected a possible problem with the location of your "_spark_metadata"
+         |directory and you likely need to move it before restarting this query.
+         |
+             |Earlier version of Spark incorrectly escaped paths when writing out the
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1357]: Please, fix indentation here and below.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1357]: Thanks you for reminder.

**Comment by beliefer** [comment]
ping @allisonwang-db

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @beliefer .

**Comment by beliefer** [comment]
@MaxGekk Thank you!

**Pull request closed without merging**